native/cocos/renderer/gfx-metal/MTLSemaphore.h

```diff
@@ -35,7 +35,12 @@ class CCMTLSemaphore final {
     explicit CCMTLSemaphore(uint32_t initialValue) : _semaphoreCount(initialValue) {
         _semaphore = dispatch_semaphore_create(initialValue);
     }
-    ~CCMTLSemaphore() = default;
+    ~CCMTLSemaphore() {
+        if (_semaphore) {
+            dispatch_release(_semaphore);
+            _semaphore = nullptr;
```

Comment:
After this change, it crashes.
![Screen Shot 2022-07-25 at 19 09 57](https://user-images.githubusercontent.com/493372/180764117-14817433-01a1-4745-8dae-278a205111bf.png)